### PR TITLE
Fixed @PendingFeature in case of exceptions

### PR DESCRIPTION
**Pull request by paplorinc**
Work In Progress, please advise proper solutions for the TODOs

### CODE REVIEW

File: spock-core/src/main/java/org/spockframework/runtime/extension/builtin/PendingFeatureInterceptor.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
   public void intercept(IMethodInvocation invocation) throws Throwable {
     try {
       invocation.proceed();
-    } catch (AssertionError e) {
+    } catch (Throwable t) {
```
Comment (leonard84) [inline spock-core/src/main/java/org/spockframework/runtime/extension/builtin/PendingFeatureInterceptor.java]: Catching `Throwable` is a bad idea, you risk swallowing Errors that indicate that the JVM failed, e.g., `OutOfMemoryError`.

**Comment by leonard84** [comment]
I'm not so sure that we need this fix `@PendingFeature` behaves like `@NotYetImplemented` in that it only handles `AssertionError`, meaning that your code does not yet comply with your expectations, not that any error is swallowed. However, this behaviour is open for discussion.

File: spock-core/src/main/java/org/spockframework/runtime/extension/builtin/PendingFeatureInterceptor.java (hunk 1 above)
Comment (paplorinc) [inline spock-core/src/main/java/org/spockframework/runtime/extension/builtin/PendingFeatureInterceptor.java]: I agree, but `invocation.proceed()` is caught like this in `AutoCleanupInterceptor`, `BaseSpecRunner`, `FailsWithInterceptor`, `FailsWithInterceptor`, `TimeoutInterceptor` and `UnitilsInterceptor` also

**Comment by paplorinc** [comment]
Do you agree that the annotated `cells can reference previous cells` feature is pending?

**Comment by leonard84** [comment]
@paplorinc I would say that `cells can reference previous cells` is a special case since it relies on not yet implemented behavior of spock itself. The annotation is targeted at "normal" subjects, that is why it does not handle exceptions in the data provider code.

**Comment by paplorinc** [comment]
You might be right that the cell reference is a special case.

However, do you consider the test I created in `https://github.com/paplorinc/spock/commit/c20e9fe10c75b358e3b20ad06ba7472450f80cda#diff-305ac92ee940c2faee974bd0dac762e9R26` also non-tipical?

**Comment by leonard84** [comment]
I've extended `@PendingFeature` to also handle exceptions, see #528 for the details.

**Comment by paplorinc** [comment]
Thanks!

**Pull request closed without merging**